internal/driver/fetch.go

```diff
@@ -41,30 +41,60 @@ import (
 // there are some failures. It will return an error if it is unable to
 // fetch any profiles.
 func fetchProfiles(s *source, o *plugin.Options) (*profile.Profile, error) {
-	sources := make([]profileSource, 0, len(s.Sources)+len(s.Base))
+	normalize := s.Normalize
+
+	sources := make([]profileSource, 0, len(s.Sources))
 	for _, src := range s.Sources {
 		sources = append(sources, profileSource{
 			addr:   src,
 			source: s,
-			scale:  1,
 		})
 	}
+
+	bases := make([]profileSource, 0, len(s.Base))
 	for _, src := range s.Base {
-		sources = append(sources, profileSource{
+		bases = append(bases, profileSource{
 			addr:   src,
 			source: s,
-			scale:  -1,
 		})
 	}
-	p, msrcs, save, cnt, err := chunkedGrab(sources, o.Fetch, o.Obj, o.UI)
+
+	// retrieve source and base profiles
+	p, pb, msrcs, mbase, save, countsrc, countbase, err := grabSourceAndBases(sources, bases, o.Fetch, o.Obj, o.UI)
+
+	// check for problems with retrieving profiles
 	if err != nil {
 		return nil, err
 	}
-	if cnt == 0 {
-		return nil, fmt.Errorf("failed to fetch any profiles")
+	if countsrc == 0 {
+		return nil, fmt.Errorf("failed to fetch any source profiles")
+	}
+	if want, got := len(sources), countsrc; want != got {
+		o.UI.PrintErr(fmt.Sprintf("fetched %d source profiles out of %d", got, want))
 	}
-	if want, got := len(sources), cnt; want != got {
-		o.UI.PrintErr(fmt.Sprintf("fetched %d profiles out of %d", got, want))
+	if want, got := len(bases), countbase; want != got {
+		o.UI.PrintErr(fmt.Sprintf("fetched %d base profiles out of %d", got, want))
+	}
+
+	// normalize by base, if necessary
+	if normalize {
+		if countbase == 0 {
```

Comment:
This won't be needed if we make grabSourcesAndBases return an error if base profiles were requested but neither were fetched.